platforms/app_link_platform_iphone.js

```diff
@@ -16,19 +16,41 @@
  *    A fallback URL, if we can't direct the user somewhere better.
  */
 /*global app_link_is_path_whitelisted*/
-function app_link_platform_iphone(platform, fallback_url) {
+function app_link_platform_iphone(platform, fallback_url, referrer) {
   fallback_url = platform.store_url || fallback_url;
   var app_url = getAppUrl();
+
+  /*
+   * Attaches the referrer to the destination URL.
+   */
+  function attachReferrer(url, referrer) {
+    var destURL = url;
+    if (destURL.indexOf('referrer') == -1 && referrer.length > 0) {
+      if (destURL.indexOf('?') > -1) {
+        destURL = destURL + '&referrer=' + referrer;
```

Comment:
There's a good chance that `document.referrer` will contain URL characters such as `&`, so we should encode it before appending:
```js
destURL = destURL + '&referrer=' + encodeURIComponent(referrer);
```

Same goes for other places where we are appending the referrer.